rules/manager.go

```diff
@@ -575,6 +588,32 @@ func (g *Group) Eval(ctx context.Context, ts time.Time) {
 			}
 		}(i, rule)
 	}
+
+	if len(g.staleSeries) != 0 {
+		app, err := g.opts.Appendable.Appender()
+		if err != nil {
+			level.Warn(g.logger).Log("msg", "creating appender failed", "err", err)
+			return
+		}
+		for _, s := range g.staleSeries {
+			// Rule that produced series no longer configured, mark it stale.
+			_, err = app.Add(s, timestamp.FromTime(ts), math.Float64frombits(value.StaleNaN))
```

Comment:
It might be nice to factor out the append and special error handling for out of order/duplicate samples to one place.

Reply:
We need different handling in different places though

Reply:
:+1: It's just a different message, but I agree it wouldn't make anything clearer.